framework/src/utils/PetscDMMoose.C

```diff
@@ -617,23 +619,54 @@ DMMooseGetEmbedding_Private(DM dm, IS * embedding)
               locator = displaced_problem->geomSearchData()._penetration_locators[it.first];
             }
             else
-            locator = dmm->_nl->_fe_problem.geomSearchData()._penetration_locators[it.first];
-
-            std::vector<dof_id_type> & slave_nodes = locator->_nearest_node._slave_nodes;
-            for (dof_id_type i = 0; i < slave_nodes.size(); ++i)
+              locator = dmm->_nl->_fe_problem.geomSearchData()._penetration_locators[it.first];
+            // penetration locator
+            std::map<dof_id_type, PenetrationInfo *>::iterator
```

Comment:
Great place for an `auto`. Also you might put the `lit` inside the for loop, it only gets evaluated once.